File: 1450-number-of-students-doing-homework-at-a-given-time/1450-number-of-students-doing-homework-at-a-given-time.js
/**
 * @param {number[]} startTime
 * @param {number[]} endTime
 * @param {number} queryTime
 * @return {number}
 */
var busyStudent = function(startTime, endTime, queryTime) {
    // Zip startTime and endTime into pairs
    const times = startTime.map((start, index) => [start, endTime[index]]);

    // Create ranges and count how many include queryTime
    const count = times
        .map(([a, b]) => Array.from({ length: b - a + 1 }, (_, i) => a + i))
        .filter(range => range.includes(queryTime))
        .length;

    return count;
};

File: 1450-number-of-students-doing-homework-at-a-given-time/1450-number-of-students-doing-homework-at-a-given-time.rb
# @param {Integer[]} start_time
# @param {Integer[]} end_time
# @param {Integer} query_time
# @return {Integer}
def busy_student(start_time, end_time, query_time)
    start_time.zip(end_time).map{|a,b| (a..b).to_a}.count{|pair| pair.include?(query_time)}
end